src/sas/qtgui/Plotting/SlicerParameters.py

```diff
@@ -289,6 +408,46 @@ def onApply(self):
             self.save1DPlotsForPlot(plots)
         pass  # debug anchor
 
+    def onDelete(self):
+        """
+        Delete the current slicer
+        """
+        # Pop up a confirmation dialog
+        reply = QtWidgets.QMessageBox.question(self, 'Delete Slicer',
+                                             'Are you sure you want to delete this slicer?',
+                                             QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
+                                             QtWidgets.QMessageBox.No)
+        if reply == QtWidgets.QMessageBox.No:
+            return
```

Comment:
Good use of defaults here! This will catch the window closing by the ESC key, or the window X.